### DISCUSSION TITLE
ReferenceError: Property 'TextEncoder' doesn't exist, js engine: hermes

### DISCUSSION THREAD

**Discussion by a943999733** [WalletConnect/walletconnect-monorepo#2861] [Developer Support] [answered]
ReferenceError: Property 'TextEncoder' doesn't exist, js engine: hermes

**Comment by boidushya** [comment, marked as answer]
Hey this is because you need TextEncoder polyfills for typescript.

Try following these steps: 
`yarn add text-encoding big-integer`

Code:
```ts
const TextEncodingPolyfill = require('text-encoding');
const BigInt = require('big-integer')

Object.assign(global, {
    TextEncoder: TextEncodingPolyfill.TextEncoder,
    TextDecoder: TextEncodingPolyfill.TextDecoder,
    BigInt: BigInt,
});
```

Add this code to your App.ts depending on your project to make sure its executed on page load

**Comment by ignaciosantise** [comment]
You can also polyfill it using our react-native-compat package

1. `yarn add @walletconnect/react-native-compat`
2. Add this line in your root app file: `import '@walletconnect/react-native-compat'`

And regarding BigInt, if you are using RN > 0.70, you don't need to polyfill it anymore as it comes with Hermes 💪
